CRM/Civicase/Hook/PermissionCheck/CaseCategory.php

```diff
@@ -123,7 +123,7 @@ private function modifyPermissionCheckForCategory($permission, &$granted, $caseC
   private function shouldRun($permission, $granted, $contactId) {
     $defaultCasePermissions = array_column($this->caseCategoryPermission->get(), 'name');
 
-    return in_array($permission, $defaultCasePermissions) && !$granted && !$contactId;
+    return in_array($permission, $defaultCasePermissions) && !$contactId;
```

Comment:
I don't think this should be changed but let me explain what happens here. 

This hook class basically runs only if there is a denial of any of the default case permissions that ships with Civicrm. If there is a denial, then we get the case category for the page/url and then check if the user has the equivalent case category permission that was denied, if the user has access to the equivalent permission, then we grant access, if not then the denial stands.

With this change, there might be problems in other places, for example, why do we need to run the hook if user already has the permission? Running it will mean that we will still do a replacement for the equivalent case category permission and do a check which might cause unexpected problems.

Reply:
Not applicable after refactoring.